lib/syntax_tree/language_server.rb

```diff
@@ -98,7 +98,7 @@ def format(source)
             character: 0
           }
         },
-        newText: SyntaxTree.format(source, print_width)
+        newText: file_extension == "haml" ? SyntaxTree::Haml.format(source, print_width) : SyntaxTree.format(source, print_width)
```

Comment:
This can be `SyntaxTree::HANDLERS[".#{file_extension}"].format(source, print_width)`